File: frontend/src/types/wellbeing.ts
export interface WellbeingAnalyzeForm {
  project_id: string
  temperature_c: number
  humidity_pct: number
  condition: string
  description?: string
  city?: string
}

export interface CareCity {
  id: string
  name: string
}

export interface WeatherSource {
  city: string | null
  provider: string | null
  observed_at: string | null
}

export interface WellbeingTip {
  id: string
  text: string
}

export interface FirstAidStage {
  stage: string
  symptoms: string
  action: string
}

export interface Facility {
  name: string
  location: string
  hours: string
  note: string
}

export interface HeatLevel {
  level: 'none' | 'yellow' | 'orange' | 'red'
  name: string
  advice: string
}

export interface WellbeingAnalysisResult {
  analysis_id: string
  project_id: string
  project_name: string
  created_at: string
  heat_level: 'none' | 'yellow' | 'orange' | 'red'
  heat_level_name: string
  advice: string
  restriction: string
  risk_index: number
  risk_tier: string
  heat_index: number | null
  uv: string
  condition: string
  temperature_c: number
  humidity_pct: number
  description: string
  reminders: WellbeingTip[]
  allowance: string
  special_groups: string
  first_aid: FirstAidStage[]
  facilities: Facility[]
  broadcast: boolean
  is_simulated: boolean
  source: string
  rules_version: string
  auto?: boolean
  weather_source?: WeatherSource | null
}

export interface WellbeingRecordSummary {
  analysis_id: string
  project_id: string
  project_name: string
  heat_level: 'none' | 'yellow' | 'orange' | 'red'
  risk_index: number
  heat_index: number | null
  is_simulated: boolean
  created_at: string
  city?: string | null
  auto?: boolean
}

export interface WeatherSnapshot {
  available: boolean
  reason: string | null
  provider: string | null
  temperature_c: number | null
  humidity_pct: number | null
  condition: string | null
  city: string | null
  observed_at: string | null
  is_simulated: boolean
}

export interface WellbeingSchedule {
  enabled: boolean
  time: string
  city: string | null
  last_run_at: string | null
  last_result: string | null
}

export interface WellbeingStatus {
  key: string
  status: string
  schedule: WellbeingSchedule
  available_endpoints: string[]
}

export interface WellbeingTips {
  version: string
  source: string
  heat_levels: HeatLevel[]
  restriction: Record<string, string>
  special_groups: string
  allowance: string
  tips: WellbeingTip[]
  first_aid: FirstAidStage[]
  facilities: Facility[]
  load_error: string
}

File: backend/app/schemas/wellbeing.py
from __future__ import annotations

from pydantic import BaseModel, Field


class WellbeingAnalyzeForm(BaseModel):
    """工友关怀分析表单：手动输入天气/环境数据，离线确定性计算。"""

    project_id: str
    temperature_c: float = Field(ge=-50, le=60)
    humidity_pct: float = Field(default=50.0, ge=0, le=100)
    condition: str = Field(default="晴", max_length=16)  # 晴/多云/阴/小雨/中雨/雷阵雨
    description: str = Field(default="", max_length=300)
    city: str | None = None  # 天气来源城市（实时天气联动时带入，作为记录溯源）


class CareCity(BaseModel):
    """工友关怀城市下拉项。"""

    id: str
    name: str


class WeatherSourceRead(BaseModel):
    """关怀分析所用天气来源（实时天气联动/定时关怀溯源）。"""

    city: str | None = None
    provider: str | None = None
    observed_at: str | None = None  # 观测/预报时间（ISO 或日期字符串）


class WellbeingTipRead(BaseModel):
    id: str
    text: str


class FirstAidStageRead(BaseModel):
    stage: str
    symptoms: str
    action: str


class FacilityRead(BaseModel):
    name: str
    location: str
    hours: str
    note: str


class HeatLevelRead(BaseModel):
    level: str  # none | yellow | orange | red
    name: str
    advice: str


class WellbeingAnalysisResponse(BaseModel):
    analysis_id: str
    project_id: str
    project_name: str
    created_at: str
    heat_level: str
    heat_level_name: str
    advice: str
    restriction: str
    risk_index: int  # 0~100
    risk_tier: str  # 低/中/高/极高
    heat_index: float | None  # 体感温度（humidex 近似）
    uv: str
    condition: str
    temperature_c: float
    humidity_pct: float
    description: str
    reminders: list[WellbeingTipRead]
    allowance: str
    special_groups: str
    first_aid: list[FirstAidStageRead]
    facilities: list[FacilityRead]
    broadcast: bool  # 是否已联动现场语音广播（达到档位且配置了 BROADCAST_WEBHOOK_URL）
    broadcast_eligible: bool = False  # 是否达到播报/蜂鸣器触发档位（不依赖 webhook 配置）
    buzzer: bool = False  # 是否已联动现场蜂鸣器硬报警（达到档位且配置了 ALERT_WEBHOOK_URL）
    is_simulated: bool
    source: str
    rules_version: str
    auto: bool = False  # 是否系统定时关怀自动生成（非手动）
    weather_source: WeatherSourceRead | None = None  # 天气来源（城市/Provider/时间）


class WellbeingRecordSummary(BaseModel):
    analysis_id: str
    project_id: str
    project_name: str
    heat_level: str
    risk_index: int
    heat_index: float | None
    is_simulated: bool
    created_at: str
    city: str | None = None  # 天气来源城市
    auto: bool = False  # 是否系统定时关怀自动生成


class WeatherRead(BaseModel):
    """实时天气查询结果：available=false 表示未配置/请求失败，前端回退手动输入。"""

    available: bool
    reason: str | None
    provider: str | None
    temperature_c: float | None = None
    humidity_pct: float | None = None
    condition: str | None = None
    city: str | None = None
    observed_at: str | None = None
    is_simulated: bool = False


class WellbeingTips(BaseModel):
    """工友关怀规则库静态内容（前端提示/急救/设施展示用）。"""

    version: str
    source: str
    heat_levels: list[HeatLevelRead]
    restriction: dict[str, str]
    special_groups: str
    allowance: str
    tips: list[WellbeingTipRead]
    first_aid: list[FirstAidStageRead]
    facilities: list[FacilityRead]
    load_error: str = ""
